ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/EeaPrivateNonceProvider.java

```diff
@@ -12,66 +12,46 @@
  */
 package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
 
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequestId;
 import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
 
-import java.io.IOException;
 import java.math.BigInteger;
 import java.util.List;
 
-import com.google.common.collect.Lists;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.web3j.protocol.Web3jService;
-import org.web3j.protocol.core.Request;
-import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
 import org.web3j.utils.Base64String;
 
 public class EeaPrivateNonceProvider implements NonceProvider {
 
-  private static final Logger LOG = LogManager.getLogger();
-
-  private final Web3jService web3jService;
+  protected static int lastId = 0;
```

Comment:
ditto as before, use AtomicInteger and remove protected.

Reply:
resolved in VertxNonceRequestTransmitter